### PR TITLE
Build dev images for arm

### PR DESCRIPTION
**Pull request by kmcnellis**
### Description

Use the base Alpine image (since eclipse-temurin doesn't build alpine for arm), and docker buildx to enable multiplatform builds for the dev & formatter docker images.
 
## Release notes:

n/a

### Checklist

- [ ] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: bin/build-dev [hunk 1]
```diff
@@ -3,9 +3,12 @@
 # DOC: Build a new development docker image and optionally push to Docker Hub if PUSH_IMAGE is set to anything.
 
 source bin/lib.sh
+export DOCKER_BUILDKIT=1
+
+PLATFORM="${PLATFORM:-"linux/amd64"}"
```
Comment (nbeloglazov) [inline bin/build-dev:8]: Can you add comment explaining what this line is doing? Is that bash string manipulation/substring?

File: prod.Dockerfile [hunk 2]
```diff
@@ -1,4 +1,5 @@
-FROM eclipse-temurin:11.0.15_10-jdk-alpine AS stage1
+# syntax=docker/dockerfile:1
+FROM --platform=$BUILDPLATFORM eclipse-temurin:11.0.15_10-jdk-alpine AS stage1
```
Comment (nbeloglazov) [inline prod.Dockerfile:2]: Given that everywhere else you are building images from scratch, would it make sense to build here one too for consistency? Or there are certain benefits in using eclipse-temurin here?

File: bin/build-dev (hunk 1 above)
Comment (kmcnellis) [reply bin/build-dev:8]: This just sets a default for the variable

File: prod.Dockerfile (hunk 2 above)
Comment (kmcnellis) [reply prod.Dockerfile:2]: I'm not entirely sure, but given that this is the production image, we do want it to be as secure as possible.  I think it's reasonable to have a dependance on the package management system in our development images, given that those are just running locally, but there's a bit of an extra layer of confidence, security, and validation in using officially vetted docker images for our production instance.  The things they do for the official images are in this dockerfile for jdk: https://github.com/adoptium/containers/blob/main/11/jdk/alpine/Dockerfile.releases.full and this for jre: https://github.com/adoptium/containers/blob/main/11/jre/alpine/Dockerfile.releases.full 

They take several steps to be more secure, like checksums, and pinning specific versions of java

**Review by azizova-leyla — APPROVED** [review]

**Pull request merged**